### PR TITLE
fix: Show Internal Error when rendering default Error page when a runtime error occurs on a 404 page

### LINKED ISSUE #10898: Incorrect error message when navigating to a non-existent route and rendering the static fallback error page
From https://github.com/sveltejs/kit/pull/10565#issuecomment-1772566464

When navigating to a non-existent route, sveltekit tries to render the default error page. However, if an error occurs during rendering (e.g., referencing an undefined property in the root `+layout.svelte` file), sveltekit returns the static fallback error page instead.

static fallback error page with a "not found" error message instead of "internal server error"

I think in this case, it's working correctly. Since the error occurred while rendering the default error page, we get the static fallback error page. However, the error message is incorrect. It should be "Internal Error" although the route we are navigating to doesn't exist.
https://github.com/sveltejs/kit/blob/fb611ef37966b5c614ed1346de17d96019bdba8e/packages/kit/src/runtime/server/utils.js#L107

### PR DESCRIPTION
**Pull request by hjhopp**
Fixes #10898 

This change will show an error message "Internal Error" when reproducing the issue discussed above.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/kit/src/runtime/server/utils.js [hunk 1]
```diff
@@ -97,17 +97,20 @@ export async function handle_fatal_error(event, options, error) {
...
-			(await options.hooks.handleError({ error, event })) ?? {
-				message: event.route.id != null ? 'Internal Error' : 'Not Found'
-			}
-		);
+	if (__SVELTEKIT_DEV__ && typeof error == 'object') {
+		fix_stack_trace(error);
 	}
+
+	const is_runtime_error =
+		error instanceof TypeError || error instanceof SyntaxError || error instanceof ReferenceError;
```
Comment (benmccann) [inline packages/kit/src/runtime/server/utils.js:107]: I'm not sure if this is an exhaustive list or if it'd be possible to create one. Perhaps we could go in the other direction? What the type of the error when a 404 is returned without additional error? Could we check if it's that type?

Comment (eltigerchino) [reply packages/kit/src/runtime/server/utils.js:107]: iirc we don’t have a type for that right now so we'd have to create one

Comment (hjhopp) [reply packages/kit/src/runtime/server/utils.js:107]: It looks like we get to this 500 error from here: https://github.com/sveltejs/kit/blob/dbbd4c7fc1ae2abef584a29a688e2247a80b9792/packages/kit/src/runtime/server/page/respond_with_error.js#L108

The 404 error happens here: https://github.com/sveltejs/kit/blob/dbbd4c7fc1ae2abef584a29a688e2247a80b9792/packages/kit/src/runtime/server/page/respond_with_error.js#L92C63-L92C63

I wondered if perhaps the error message should get handled in that line 108 in response_with_error.

**Comment by eltigerchino** [comment]
These are the two places we are throwing a 404 error internally.
The server: https://github.com/sveltejs/kit/blob/5eb287900f804db47c5c40b211a3a75bb3f67f4f/packages/kit/src/runtime/server/respond.js#L483
and the client: https://github.com/sveltejs/kit/blob/5eb287900f804db47c5c40b211a3a75bb3f67f4f/packages/kit/src/runtime/client/client.js#L1009

We should also update the client-side error handling that is using a similar logic as the server one we're changing now:
https://github.com/sveltejs/kit/blob/5eb287900f804db47c5c40b211a3a75bb3f67f4f/packages/kit/src/runtime/client/client.js#L1330

**Comment by hjhopp** [comment]
Would something like making those 404 errors a specific type of error, and then checking that the error is that type in `handle_error_and_jsonify`, only returning "Not Found" if both the route ID is null and the error is a 404 error work?

**Review by dummdidumm — APPROVED** [review]
Thank you!

**Pull request merged**